### PR TITLE
Swarm limor

### PR DESCRIPTION
**Pull request by limorgottlieb**

### CODE REVIEW

**Comment by tiagoalves83** [comment]
We did similar work (https://github.com/spotify/docker-client/pull/723).
Are you working on the docker v2 plugin api ? Do you need help ?

**Comment by limorgottlieb** [comment]
@tiagoalves83 - I see it now. I started working on it 2 weeks ago and there was no PR for that.
I'm not working on the docker v2 plugin API. I'm using this docker java library and need the swarm functionality to be part of the library.

**Comment by limorgottlieb** [comment]
Can one of the admins review this PR?
I need the swarm functionality to be merged.
Thanks!

File: .travis.sh [hunk 1]
```diff
@@ -40,6 +40,45 @@ case "$1" in
       sudo mount -t mqueue none /dev/mqueue
     fi
 
+    if [[ "$DOCKER_VERSION" =~ ^1\.6\..* ]]; then
+      # docker-engine 1.6.x packages don't seem to have the upstart job config,
+      # so write it ourselves
+      echo '
```
Comment (mattnworb) [inline .travis.sh]: using a here-doc here would mean you don't have to worry about escaping the single-quotes below

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 2]
```diff
@@ -10,9 +10,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
+ *
  *      http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: please avoid unnecessary whitespace changes; this is likely to break the license-plugin check in the build

File: src/main/java/com/spotify/docker/client/messages/swarm/SwarmInitRequest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+@JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
+public class SwarmInitRequest {
```
Comment (mattnworb) [inline src/main/java/com/spotify/docker/client/messages/swarm/SwarmInitRequest.java]: please use autovalue for message classes like this

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 4]
```diff
@@ -305,7 +309,10 @@
   @Before
   public void setup() throws Exception {
     registryAuth = RegistryAuth.builder().username(AUTH_USERNAME).password(AUTH_PASSWORD).build();
-    final DefaultDockerClient.Builder builder = DefaultDockerClient.fromEnv();
+    //    final DefaultDockerClient.Builder builder = DefaultDockerClient.fromEnv();
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: please avoid committing commented-out code

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 5]
```diff
@@ -305,7 +309,10 @@
...
+    final DefaultDockerClient.Builder builder = DefaultDockerClient.builder()
+            .uri("http://10.151.17.226:4550");
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: what is this IP address?

**Comment by davidxia** [comment]
Addressed by https://github.com/spotify/docker-client/pull/729
